code/lib/preview-api/src/modules/preview-web/render/StoryRender.ts

```diff
@@ -226,10 +239,11 @@ export class StoryRender<TRenderer extends Renderer> implements Render<TRenderer
 
       if (abortSignal.aborted) return;
 
-      await this.runPhase(abortSignal, 'rendering', async () => {
-        const teardown = await this.renderToScreen(renderContext, canvasElement);
-        this.teardownRender = teardown || (() => {});
-      });
+      const isMountDestructured = playFunction && mountDestructured(playFunction);
```

Comment:
Maybe we should store this property on the prepared story rather than checking in multiple places?

Reply:
I am working on a new PR that fixes this.